Source/WebCore/css/CSSPrimitiveValue.cpp

```diff
@@ -1212,24 +1212,43 @@ String CSSPrimitiveValue::stringValue() const
     }
 }
 
+static NEVER_INLINE ASCIILiteral formatNonfiniteValuePrefix(double number)
+{
+    if (number == std::numeric_limits<double>::infinity())
+        return "infinity"_s;
+    if (number == -std::numeric_limits<double>::infinity())
+        return "-infinity"_s;
+    ASSERT(std::isnan(number));
+    return "NaN"_s;
+}
+
+static NEVER_INLINE void formatNonfiniteValue(StringBuilder& builder, double number, ASCIILiteral suffix)
+{
+    return builder.append(formatNonfiniteValuePrefix(number), suffix.isEmpty() ? ""_s : " * 1"_s, suffix);
```

Comment:
Kinda wish we could share the argument list between the `StringBuilder` and `makeString` versions.